### PR TITLE
Fix cross-attention

### PR DESCRIPTION
**Pull request by jbochi**
## Proposed changes

With the current code, ln2 is a no-op. Its output should be passed to the cross-attention layer

## Checklist

Put an `x` in the boxes that apply.

- [x] I have read the [CONTRIBUTING](https://github.com/ml-explore/mlx/blob/main/CONTRIBUTING.md) document
- [ ] I have run `pre-commit run --all-files` to format my code / installed pre-commit prior to committing changes
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have updated the necessary documentation (if needed)

### CODE REVIEW

File: python/mlx/nn/layers/transformer.py [hunk 1]
```diff
@@ -157,7 +157,7 @@ def __call__(self, x, memory, x_mask, memory_mask):
         x = x + y
 
         y = self.ln2(x)
-        y = self.cross_attention(x, memory, memory, memory_mask)
```
Comment (awni) [inline python/mlx/nn/layers/transformer.py:160]: Great catch! Thank you! This little typos are often the hardest to find :).

**Review by awni — APPROVED** [review]
@jbochi Would you like to add your name to our [contributors section](https://github.com/ml-explore/mlx/blob/main/ACKNOWLEDGMENTS.md#individual-contributors)? We just added it

File: python/mlx/nn/layers/transformer.py (hunk 1 above)
Comment (jbochi) [reply python/mlx/nn/layers/transformer.py:160]: 💯

**Comment by jbochi** [comment]
> @jbochi Would you like to add your name to our [contributors section](https://github.com/ml-explore/mlx/blob/main/ACKNOWLEDGMENTS.md#individual-contributors)? We just added it

What an honor to be the first contributor on the list. :D 

BTW, I'm super impressed by MLX. Amazing work!

**Review by awni — APPROVED** [review]

**Pull request merged**